common-http/src/main/java/io/cdap/common/http/HttpResponse.java

```diff
@@ -80,6 +134,26 @@ public String getResponseBodyAsString(Charset charset) {
     return builder.build();
   }
 
+  private byte[] getResponseBodyFromStream() {
+    try {
+      if (inputStream == null) {
+        throw new IOException("Input stream has already been consumed");
+      }
+      return ByteStreams.toByteArray(inputStream);
+    } catch (IOException e) {
```

Comment:
Why not propagate out? The constructor who call this method has throws IOException.

Reply:
fixed